l2.py

```diff
@@ -0,0 +1,234 @@
+"""
+This module defines functions related to spinning a devnet op-geth node.
+"""
+
+import os
+import shutil
+import subprocess
+import sys
+
+import libroll as lib
+from paths import OPPaths
+from processes import PROCESS_MGR
+
+
+####################################################################################################
+
+DEVNET_L2_DATA_DIR = "db/devnetL2"
+"""Directory to store the devnet op-geth blockchain data."""
+
+
+####################################################################################################
+
+def deploy_l2_devnet(paths: OPPaths):
+    """
+    Spin the devnet op-geth node, doing whatever tasks are necessary, including installing op-geth,
+    generating the genesis file and config files.
+    """
+    generate_devnet_l2_genesis(paths)
+    start_devnet_l2_node(paths)
+    print("Devnet L2 deployment is complete! L2 node is running.")
+    PROCESS_MGR.wait_all()
+
+
+####################################################################################################
+
+def generate_devnet_l2_genesis(paths: OPPaths):
+    """
+    Generate the L2 genesis file and rollup configs.
+    """
+    if os.path.exists(paths.l2_genesis_path):
+        print("L2 genesis and rollup configs already generated.")
+    else:
+        print("Generating L2 genesis and rollup configs.")
+        try:
+            lib.run(
+                "generating L2 genesis and rollup configs",
+                ["go", "run", "cmd/main.go", "genesis", "l2",
+                    "--l1-rpc=http://localhost:8545",
+                    f"--deploy-config={paths.network_config_path}",
+                    f"--deployment-dir={paths.devnet_l1_deployment_dir}",
+                    f"--outfile.l2={paths.l2_genesis_path}",
+                    f"--outfile.rollup={paths.rollup_config_path}"],
+                cwd=paths.op_node_dir)
+        except Exception as err:
+            raise lib.extend_exception(err, prefix="Failed to generate L2 genesis and rollup configs: ")
+
+
+####################################################################################################
+
+
+class DevnetL2Config:
+    def __init__(self, geth_data_dir: str, paths: OPPaths):
+        self.data_dir = geth_data_dir
+        """Geth data directory for devnet op-geth node."""
+
+        self.keystore_dir = f"{self.data_dir}/keystore"
+        """Keystore directory for devnet op-geth node (each file stores an encrypted signer key)."""
+
+        self.chaindata_dir = f"{self.data_dir}/geth/chaindata"
+        """Directory storing chain data."""
+
```

Comment:
Might want to remove some options here that become unused as a result of change in options highlighted in comments on the `start_devnet_l2_node` function.